keanu-project/src/main/java/io/improbable/keanu/algorithms/NetworkSamples.java

```diff
@@ -78,8 +81,9 @@ public NetworkSamples downSample(final int downSampleInterval) {
                 e -> downSample(e.getValue(), downSampleInterval)
                 )
             );
+        final List<Double> withLogProbsDownSampled = (List<Double>) downSample(logOfMasterPForEachSample, downSampleInterval);
```

Comment:
Looks like you've repurposed the `downSample(...)` method to down sample probabilities as well. Good work! Can we make the method generic? It might require a cast on `getDoubleTensorSamples(...)` and `getIntegerTensorSamples(...)` but that's where I would expect the cast instead of here with `withLogProbsDownSampled `.